Workflows context

laclac102 opened the pull request:
This PR aligns the dashboard UI with the [UX design here](https://xd.adobe.com/view/cfc2d703-cc19-4a66-b736-0b9f2e655fae-9a91/), updates workflow data handling, and enables dynamic form rendering based on pipeline schema (following the [nf-core nextflow_schema.json format](https://nextflow-io.github.io/nf-validation/nextflow_schema/nextflow_schema_specification/#required-parameters))

**Changes Included**

- Updated dashboard menu structure to align with the UX design
- Update stepper for workflow launcher
- Refactored WorkflowsContext to support schema-based workflows
- Updated the workflow database model to store schema URLs for pipelines
<img width="854" alt="Screenshot 2025-07-07 at 4 32 55 pm" src="https://github.com/user-attachments/assets/70994733-3320-4e27-92ae-0165943c04e6" />

- Modified TypeScript interfaces to support schema fields
- Enabled UI to render dynamic input forms based on pipeline schema
- Schema rendering supports required fields and follows the structure of nf-core's nextflow_schema.json ($defs.input_output_options/definitions.input_output_options)
<img width="1300" alt="Screenshot 2025-07-07 at 4 30 09 pm" src="https://github.com/user-attachments/assets/f769de5b-b405-4d4c-a781-149b2d802e28" />

 **How to test this UI manually**
`npm run dev`
For workflow execution testing, please contact me to get .env.local

amandazhuyilan commented:
Hi @marius-mather - can you recommend any of the following to be added to this repo:
- unit tests framework
- sample github actions to run the unit tests
- linters

marius-mather commented:
For the unit test framework I like `vitest`, there's a guide [in the Next.js documentation](https://nextjs.org/docs/app/guides/testing/vitest) for setting it up (may need the "manual configuration" option as the project has already been created).

If you want to go down the path of end-to-end testing [Cypress](https://nextjs.org/docs/app/guides/testing/cypress) is also good - but it will be extra work, so maybe start with the unit tests for now.

I'm not sure if there's a good prepackaged Github Action to run the tests, but [this](https://stevekinney.com/courses/testing/continuous-integration) looks like a good starting point: you should be able to just create an action that runs `npm install` and then `npm test` to run the tests.

For linting, NextJS has some [built in linting](https://nextjs.org/docs/app/api-reference/config/eslint) that would be a good place to start, and you can create a GitHub Action that runs the linter.

marius-mather reviewed (changes requested):
Looking great so far - a couple of minor things to clean up, but otherwise I think this is looking like a great start

laclac102 commented:
### Updates in This PR
Hihi @marius-mather @amandazhuyilan , many thanks for the above comments, i've addressed the request changes. Below are all changes and updates in 4 latest commits:

- Feature: Introduced the stepper component for multi-step workflow submission
- Fix: Launch button now submits the form correctly and triggers workflow launch
- Test: Added unit test  for the Home page and linting
- Chore: Fixed a typo in the codebase

Kindly take another look when you have a moment and please let me know if anything else is needed 
***Note: *** CI in [this PR](https://github.com/Australian-Structural-Biology-Computing/structural-bio-portal/pull/2)

laclac102 commented:
Many thanks to youuuu @marius-mather
